frontend/src/state/apiSlice.ts

```diff
@@ -715,6 +718,118 @@ export const boardsApi = createApi({
         { type: "Columns", id: "LIST" },
         { type: "Ticket", id: "LIST" }
       ]
+    }),
+
+    getScopes: builder.query<Scope[], string>({
+      query: (boardId) => ({
+        url: `scopes/${boardId}`,
+        method: "GET"
+      }),
+      providesTags: [{ type: "Scopes", id: "LIST" }]
+    }),
+
+    addScope: builder.mutation<Scope, { boardId: string; title: string }>({
+      query: ({ boardId, title }) => ({
+        url: `scopes/${boardId}/`,
+        method: "POST",
+        body: { title }
+      }),
+      invalidatesTags: () => invalidateRemoteCache(["Scopes"])
+    }),
+
+    deleteScope: builder.mutation<Scope, { boardid: string; scopeid: string }>({
+      query: ({ boardid, scopeid }) => ({
+        url: `scopes/${boardid}/`,
+        method: "DELETE",
+        body: { scopeid }
+      }),
+      invalidatesTags: () => invalidateRemoteCache(["Ticket", "Scopes"])
+    }),
+
+    setDoneColumns: builder.mutation<Scope, { scopeid: string; columnidlist: string[] }>({
+      query: ({ scopeid, columnidlist }) => ({
+        url: `scopes/${scopeid}/set_done_columns`,
+        method: "POST",
+        body: { done_columns: columnidlist }
+      }),
+      invalidatesTags: () => invalidateRemoteCache(["Scopes"])
+    }),
+
+    setScopeForecast: builder.mutation<Scope, { scopeid: string }>({
+      query: ({ scopeid }) => ({
+        url: `scopes/${scopeid}/set_scope_forecast`,
+        method: "POST"
+      }),
+      invalidatesTags: () => invalidateRemoteCache(["Scopes"])
+    }),
+
+    setScopeTitle: builder.mutation<{ success: boolean }, { scopeid: string; title: string }>({
+      query: ({ scopeid, title }) => ({
+        url: `scopes/${scopeid}/set_title`,
+        method: "POST",
+        body: { title: title }
+      }),
+      invalidatesTags: () => invalidateRemoteCache(["Ticket", "Scopes"])
+    }),
+
+    addTaskToScope: builder.mutation<{ success: boolean }, { scope: SimpleScope; ticketid: string }>({
+      query: ({ scope, ticketid }) => ({
+        url: `scopes/${scope.scopeid}/tickets`,
+        method: "POST",
+        body: { ticketid }
+      }),
+      //update optimistically
+      onQueryStarted({ ticketid, scope }, apiActions) {
+        const invalidationTags: CacheInvalidationTag[] = [{ type: "Ticket", id: ticketid }]
+        updateCache(
+          "getTaskListByColumnId",
+          invalidationTags,
+          (draft) => {
+            const tasks = draft as Task[]
+            const task = tasks.find((task) => task.ticketid === ticketid)
+            if (task) {
+              task.scopes.push({ scopeid: scope.scopeid, title: scope.title })
+            }
+          },
+          apiActions
+        )
+
+        apiActions.queryFulfilled.finally(() => {
+          invalidateRemoteCache(invalidationTags)
+          boardsApi.util.invalidateTags(invalidationTags)
+        })
+      },
+      invalidatesTags: () => invalidateRemoteCache(["Scopes"])
```

Comment:
```suggestion

        invalidationTags.push("Scopes")

        apiActions.queryFulfilled.finally(() => {
          invalidateRemoteCache(invalidationTags)
          apiActions.dispatch(boardsApi.util.invalidateTags(invalidationTags))
        })
      }
```

Sama juttu kun alla.

Reply:
Korjattu https://github.com/Futuboard/Futuboard/pull/73/commits/cab5a6d77d35620b89a2379cce6bd409b27beac5